### PR TITLE
Delete unnecessary judgments for deploymentController

### PR DESCRIPTION
**Pull request by mysunshine92**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/controller/deployment/sync.go [hunk 1]
```diff
@@ -454,7 +454,7 @@ func (dc *DeploymentController) cleanupDeployment(ctx context.Context, oldRSs []
 	for i := int32(0); i < diff; i++ {
 		rs := cleanableRSes[i]
 		// Avoid delete replica set with non-zero replica counts
-		if rs.Status.Replicas != 0 || *(rs.Spec.Replicas) != 0 || rs.Generation > rs.Status.ObservedGeneration || rs.DeletionTimestamp != nil {
+		if rs.Status.Replicas != 0 || *(rs.Spec.Replicas) != 0 || rs.Generation > rs.Status.ObservedGeneration {
```
Comment (soltysh) [inline pkg/controller/deployment/sync.go:457]: Why do you want to remove a condition that ensure we ignore replicasets which are already marked for deletion?

**Review by soltysh — CHANGES_REQUESTED** [review]
/hold

**Comment by dims** [comment]
this PR has been on hold for about 2 weeks. Can you please see how to unblock this by reviewing comments etc? or please close this out (Trying to declutter our open list of PRs)

**Comment by soltysh** [comment]
/close
haven't heard from the author

**Pull request closed without merging**